tests/unit_tests/test_maps.py

```diff
@@ -150,45 +196,87 @@ def test_maps_inject_sql_sort(test_client):
+
+    assert json.loads(response.get_data(as_text=True))['message'] == "Value 'is_ranked' is missing"
+
+
+def test_maps_upload_ranked(oauth, app, maps, tmpdir):
+    _test_map_upload(oauth, app, maps, tmpdir, True)
 
-    with open(upload_dir.join('map_name.zip').strpath, 'r') as file:
-        assert file.read() == 'my file contents'
 
+def test_maps_upload_unranked(oauth, app, maps, tmpdir):
+    _test_map_upload(oauth, app, maps, tmpdir, False)
 
-def test_maps_upload_no_file_results_400(test_client, app, tmpdir):
-    response = test_client.post('/maps/upload')
+
+def test_maps_upload_no_file_results_400(oauth, app, tmpdir):
+    response = oauth.post('/maps/upload')
 
     assert response.status_code == 400
     assert json.loads(response.get_data(as_text=True))['message'] == 'No file has been provided'
 
 
-def test_maps_upload_txt_results_400(test_client, app, tmpdir):
-    response = test_client.post('/maps/upload', data={'file': (BytesIO('1'.encode('utf-8')), 'map_name.txt')})
+def test_maps_upload_txt_results_400(oauth, app, tmpdir):
+    response = oauth.post('/maps/upload', data={'file': (BytesIO('1'.encode('utf-8')), 'map_name.txt'),
+                                                'is_ranked': 'true'})
 
     assert response.status_code == 400
     assert json.loads(response.get_data(as_text=True))['message'] == 'Invalid file extension'
 
 
 def test_map_by_name(test_client, app, maps):
-    response = test_client.get('/maps?filter%5Btechnical_name%5D=b.v0001')
+    response = test_client.get('/maps?filter%5Btechnical_name%5D=scmp_002.v0001')
 
     assert response.status_code == 200
     assert response.content_type == 'application/vnd.api+json'
 
     result = json.loads(response.data.decode('utf-8'))
     assert 'data' in result
-    assert result['data']['id'] == 222
-    assert result['data']['attributes']['display_name'] == 'b'
-    assert result['data']['attributes']['download_url'] == 'http://content.faforever.com/faf/vault/maps/b.v0001.zip'
+    assert result['data']['id'] == 2
+    assert result['data']['attributes']['display_name'] == 'SCMP_002'
+    assert result['data']['attributes'][
+               'download_url'] == 'http://content.faforever.com/faf/vault/maps/scmp_002.v0001.zip'
     assert result['data']['attributes']['thumbnail_url_small'] == 'http://content.faforever.com/faf/vault' \
-                                                                  '/map_previews/small/maps/b.v0001.zip'
+                                                                  '/map_previews/small/maps/scmp_002.v0001.zip'
     assert result['data']['attributes']['thumbnail_url_large'] == 'http://content.faforever.com/faf/vault' \
-                                                                  '/map_previews/large/maps/b.v0001.zip'
+                                                                  '/map_previews/large/maps/scmp_002.v0001.zip'
+
+
+def _test_map_upload(oauth, app, maps, tmpdir, ranked):
```

Comment:
I see. To make py.test not pick it up since it's not a test function, but a helper.

You should make a `ranked` fixture instead that has two values; true and false.